internal/stores/hostdb.go

```diff
@@ -133,15 +135,24 @@ func (db *SQLStore) Host(hostKey consensus.PublicKey) (hostdb.Host, error) {
 
 // Hosts returns up to max hosts that have not been interacted with since
 // the specified time.
-func (db *SQLStore) Hosts(notSince time.Time, max int) ([]hostdb.Host, error) {
-	// Filter all hosts for the ones that have not been updated since a
-	// given time.
+func (db *SQLStore) Hosts(offset, limit int, whitelist []string) ([]hostdb.Host, error) {
+	if offset < 0 {
+		return nil, ErrNegativeOffset
+	}
+
 	var fullHosts []dbHost
-	err := db.db.Table("hosts").
-		Joins("LEFT JOIN host_interactions ON host_interactions.db_host_id = hosts.ID").
-		Group("Public_Key").
-		Having("IFNULL(MAX(Timestamp), 0) < ?", notSince.UTC()). // use UTC since we stored timestamps in UTC
-		Limit(max).
+	tx := db.db.Model(&dbHost{}).
+		Joins("LEFT JOIN announcements ann ON hosts.id = ann.db_host_id")
+
+	if len(whitelist) > 0 {
+		tx = tx.Where("ann.net_address IN (?)", whitelist)
```

Comment:
As mentioned before I think this doesn't scale as well as a dedicated column which we may want to introduce to explicitly ignore hosts. We could allow for setting that column using different means such as pubkeys, IPs or URLs. 

Primarily I'd try to speed up the query though. e.g. by not preloading all interactions and announcements anymore.